### PR TITLE
Remove AwaitingReview view, give reviewer access to submission all

### LINKED ISSUE #4146: Remove obsolete awaiting_review view
Reviewers then need access to the new submission all view. That view must be made to take reviewer settings in to account.

### PR DESCRIPTION
**Pull request by frjo**
Fixes #4146

- [ ] Rename views all beta to views all.
- [ ] Give reviewers access to the submission all view.
- [ ] Make submission all list only reviews reviewers have access to, including reviewers settings if set.
- [ ] Remove batch actions and review filter for reviewers.
- [ ] Remove old AwaitingReview view.
- [ ] Implement show_applicant_identity for submission all view.

## Test Steps

 - [ ] Make sure reviewers only see submission they should have access to in submission all view.

### CODE REVIEW

File: hypha/apply/dashboard/templates/dashboard/includes/submissions-waiting-for-review.html [hunk 1]
```diff
@@ -12,7 +12,7 @@ <h2 class="text-xl mb-2">
 
     {% if display_more %}
         <div class="all-submissions-table__more">
-            <a href="{% url 'apply:submissions:awaiting_review' %}">{% trans "Show all" %}</a>
+            <a href="{{ awaiting_reviews.url }}?reviewers={{ request.user.pk }}">{% trans "Show all" %}</a>
```
Comment (theskumar) [inline hypha/apply/dashboard/templates/dashboard/includes/submissions-waiting-for-review.html]: This doesn't work, I believe this should go to submission all page, with something like `?query=reviewers:@me is:open`, keeping the query intensionally composable. 


Screenshot 2024-11-12 at 7  02 53@2x

File: hypha/apply/dashboard/templates/dashboard/reviewer_dashboard.html [hunk 2]
```diff
@@ -43,7 +43,7 @@ <h3 class="text-xl mb-2">Your previous reviews</h3>
                 {% include "submissions/partials/submissions-inline.html" with submissions=my_reviewed.table.data row_layout="table" %}
 
                 <div class="border-x border-b flex items-center justify-center py-3 font-semibold">
-                    <a href="{{ my_reviewed.url }}?reviewers={{ request.user.pk }}">{% trans "Show all" %}</a>
+                    <a href="{{ my_reviewed.url }}?reviewers={{ request.user.pk }}&reviewedby=me">{% trans "Show all" %}</a>
```
Comment (theskumar) [inline hypha/apply/dashboard/templates/dashboard/reviewer_dashboard.html]: I would recommend coming up with another language in the query, something on the line of: 

`reviewed-by:@me` and `reviewed-by:<user_id>`

File: hypha/apply/funds/templates/funds/includes/submission-table-row.html [hunk 3]
```diff
@@ -1,4 +1,4 @@
-{% load i18n querystrings %}
+{% load i18n workflow_tags querystrings %}
```
Comment (theskumar) [inline hypha/apply/funds/templates/funds/includes/submission-table-row.html]: can we use `{% load show_applicant_identity from workflow_tags %}` to make it more explicit?

File: hypha/apply/funds/views_all.py [hunk 4]
```diff
@@ -111,6 +115,15 @@ def submission_all_beta(
...
 
+    if request.user.is_reviewer:
+        reviewer_settings = ReviewerSettings.for_request(request)
+        if reviewer_settings.use_settings:
+            qs = qs.for_reviewer_settings(request.user, reviewer_settings)
+        elif request.GET.get("reviewedby", False):
+            qs = qs.reviewed_by(request.user)
+        else:
+            qs = qs.in_review_for(request.user)
+
```
Comment (theskumar) [inline hypha/apply/funds/views_all.py]:
```suggestion
        else:
            qs = qs.in_review_for(request.user)

    if "reviewed-by" in search_filters:
        if "@me" in search_filters["reviewed-by"]:
            qs = qs.reviewed_by(request.user)

```

Comment (theskumar) [reply hypha/apply/funds/views_all.py]: Make `reviewed-by:@me` an independent filter.

**Review by theskumar — CHANGES_REQUESTED** [review]

File: hypha/apply/funds/templates/submissions/all.html [hunk 5]
```diff
@@ -87,7 +87,7 @@
             </a>
             <a
                 href="{% url "apply:submissions:list" %}?query=reviewer:@me"
-                href="{% url "apply:submissions:list" %}?query=reviewer:@me"
+                hx-get="{% url "apply:submissions:list" %}?query=reviewer:@me"
```
Comment (frjo) [inline hypha/apply/funds/templates/submissions/all.html:102]: @theskumar Looked like the double "href" was a mistake but it still worked?

**Comment by frjo** [comment]
@theskumar Implemented your suggestion and made some other changes. Please take a new look.

**Comment by wes-otf** [comment]
Also could we swap `overflow-x-scroll` -> `overflow-x-auto` here?:

https://github.com/HyphaApp/hypha/blob/9aca48b043ac4c11b8814458e9e7af2239847a27/hypha/apply/funds/templates/submissions/partials/submissions-inline.html#L11

that'll get rid of that gap between the table & show all

**Comment by wes-otf** [comment]
Other than those two small things this is good to go! A lot more slick!

**Pull request merged**